### DISCUSSION TITLE
Change models url with slug for better SEO

### DISCUSSION THREAD

**Discussion by manuelurenah** [civitai/civitai#29] [Ideas]
Right now the model url structure is as follow `/models/{id}` which is not great for SEO since it will generate the following `/models/1234`. To improve this, it'd be best to slugify the model title and set that as the url `/models/{slug}` (e.g.: `/model/arcane-diffusion`).

Also, we should strip any html tags from the meta description

**Comment by JustMaier** [comment]
As a quick hack for this, we could probably just make it so that the path is `/models/:id/:slug` that way we can still use the `id` to get what we need and the slug can just be for appearance, just like how reddit does it.

That said, implementing a simple slug system probably won't be that difficult. It's building a feature complete slug system that can get time consuming and hard to work with (speaking from experience).

**Comment by manuelurenah** [comment]
@JustMaier I will be implementing this as you suggested keeping the id in the url as well. Let's keep it simple then work our way as we see fit

**Comment by JustMaier** [comment]
@manuelurenah thanks for implementing this! Happy to have this live.

![image](https://user-images.githubusercontent.com/607609/204434829-82f3048d-dca3-4f61-ba68-9a3319794de2.png)

**Comment by Gogo1951** [comment]
Google has stated (like a while back now) that they really don't care what's in a URL.

Remember when they tried to kill it off all together, and hide the URL bar from showing anything more than just the top level domain in Chrome? It didn't take, but they've been trying to diminish people looking at URLs ever since.

A lot of SEO places still harp on this 10+ year-old "put keywords in the URL" but it's not true.

Look at their advice around localized URLs, and how they've said, "Keywords in the URL don't matter." 

Look at Moz's findings, that keywords in URLs can be good for users clicking them if they read the whole URL in a link somewhere... but past that, "Meh."

Look at sites like IMDB, certainly very popular, which doesn't use any keywords in the URL.

https://www.imdb.com/title/tt0076729/

Anyway just wanted to point this out. All the SEO stuff is still 99% BS. (=
